### ISSUE TITLE
Organization ID for third party API in OpenAI format?

### ISSUE THREAD

**Issue by 0xzz** [logancyang/obsidian-copilot#836]
**Is your feature request related to a problem? Please describe.**
My institution provides third party API in the OpenAI format. They require "Organization ID" to keep track of usage from different units. While it's possible specify Organization ID for the official OpenAI models, it does not seem possible to do so for third party providers.

**Describe the solution you'd like**
Allow specifying "Organization ID" when adding custom third-party provider

**Describe alternatives you've considered**
Unsure if it's possible to directly specify Organization ID in the base query string

**Additional context**
N/A

**Comment by Emt-lin** [comment]
@0xzz In next version(v2.7.3), You can select the `OpenAI` provider and fill in the `ID` at the arrow in the image below.

![CleanShot 2024-11-22 at 13 58 17](https://github.com/user-attachments/assets/8ad93bf8-23b3-4180-8387-9d468b186ae8)

**Comment by logancyang** [comment]
Thanks for reporting this, and sorry for the slow reply.

Copilot v3 is in code freeze, so we're closing this as not planned. v4 is a ground-up rewrite and is launching soon, with a very different architecture around agent mode.

If this is still an issue for you in v4 once it lands, please open a new issue and we'll take a fresh look. 🙏
